### PR TITLE
CB-5945 Enforce authz annotation on specific fields of request object

### PR DESCRIPTION
**Pull request by foldik**

### CODE REVIEW

File: authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java [hunk 1]
```diff
@@ -62,18 +124,239 @@ public static void testIfControllerClassHasAuthorizationAnnotation() {
...
+
+    private static Function<Method, Optional<String>> noParamsWith(Class<? extends Annotation>... annotations) {
+        return method -> {
+            for (Parameter parameter : method.getParameters()) {
+                for (Class<? extends Annotation> annotation : annotations) {
+                    if (parameter.isAnnotationPresent(annotation)) {
+                        Optional.of(invalid(method, '@' + annotation.getSimpleName() + " annotation is not allowed here"));
+                    }
+                }
+            }
```
Comment (horadla23) [inline authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java:229]: is it possible to replace these with stream methods?

File: authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java [hunk 2]
```diff
@@ -62,18 +124,239 @@ public static void testIfControllerClassHasAuthorizationAnnotation() {
...
+                    for (Generic generic : genericTypes) {
+                        if (generic.wrapperType.isAssignableFrom(parameter.getType())) {
+                            Type[] actualTypeArguments = ((ParameterizedType) parameter.getParameterizedType()).getActualTypeArguments();
+                            if (actualTypeArguments.length == 1 && actualTypeArguments[0].equals(generic.genericType)) {
+                                count++;
+                            }
+                        }
+                    }
+                }
+            }
```
Comment (horadla23) [inline authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java:248]: same here

File: authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java [hunk 3]
```diff
@@ -62,18 +124,239 @@ public static void testIfControllerClassHasAuthorizationAnnotation() {
...
+                Type[] actualTypeArguments = ((ParameterizedType) field.getGenericType()).getActualTypeArguments();
+                if (actualTypeArguments.length == 1 && actualTypeArguments[0].equals(generic.genericType)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static class Generic {
```
Comment (horadla23) [inline authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java:334]: can we extract this into different class with more specific name?

File: authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java [hunk 4]
```diff
@@ -62,18 +124,239 @@ public static void testIfControllerClassHasAuthorizationAnnotation() {
...
+                }
+            }
+            String errorMessageCommon = " method parameter with @" + annotation.getSimpleName() + " annotation and type " + type.getSimpleName();
+            if (count == 0) {
+                return Optional.of(invalid(method, "Misssing" + errorMessageCommon));
+            } else if (count > 1) {
+                return Optional.of(invalid(method, "Multiple" + errorMessageCommon));
+            } else {
+                return Optional.empty();
+            }
```
Comment (horadla23) [inline authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java:217]: as I see this code block appears twice in the class, please extract it

File: authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java (hunk 1 above)
Comment (foldik) [reply authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java:229]: yes

File: authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java (hunk 2 above)
Comment (foldik) [reply authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java:248]: yes

File: authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java (hunk 3 above)
Comment (foldik) [reply authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java:334]: yep

File: authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java (hunk 4 above)
Comment (foldik) [reply authorization-common/src/test/java/com/sequenceiq/authorization/EnforceAuthorizationLogicsUtil.java:217]: ok

**Pull request merged**